### PR TITLE
Feature/Create New Edit Workout Set Dialog

### PR DESCRIPTION
**Pull request by Alex-A-Fit**
## Bug/Feature description
Create Full-Screen Dialog to edit already inputted workouts

## Code Validations 
[ ] PR is under 400 lines of code Insertion and Deletion  
[X] Files are Formatted and Optimized  
[X] Clean Architecture followed  
[X] No unneccessary comments or TODOs left in codebase  
[ ] Unit Tests Added  
[ ] Unit Tests NOT added and explanation why under exemption description  
[ ] UI Tests Added  
[ ] UI Tests NOT added and explanation why under exemption description  
[ ] End to End Tests Added  
[ ] End to End Tests NOT added and explanation why under exemption description  
[X] Screenshot of UI added  

## Exemption Definition
Explain if necessary the reason for any exemptions

## UI Screenshots 

Edit_Workout_Set_Dailog_1

Edit_Workout_Set_Dialog_2

Edit_Workout_Set_Dialog_3

## Test Case Screenshots
Place test Success creations here

### CODE REVIEW

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/EditWorkoutSetDialog.kt [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                    .verticalScroll(rememberScrollState()),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                EditWorkoutSetProperties(
+                    workoutTypeDialog = workoutTypeDialog,
+                    isWorkoutValid = { isValidWorkout, workout ->
+                        isWorkoutValid = isValidWorkout
+                        if (workout != null) {
+                            newUpdatedWorkout = workout
+                        }
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/EditWorkoutSetDialog.kt]:
```suggestion
                       workout?.let {
                            newUpdatedWorkout = it
                        }
```

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetCalisthenics.kt [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (setValue == "0") {
+                    isSetErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isSetErrorVisible = true
+            }
+        },
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetCalisthenics.kt:134]: This is duplicate code. Can we extract this out into a helper function and reuse it?

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (setValue == "0") {
+                    isSetErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isSetErrorVisible = true
+            }
+        },
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt:117]: This is duplicate code. Can we extract this out into a helper function and reuse it?

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetCalisthenics.kt [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (repValue == "0") {
+                    isRepErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isRepErrorVisible = true
+            }
+        },
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetCalisthenics.kt:169]: This is duplicate code. Can we extract this out into a helper function and reuse it?

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (repValue == "0") {
+                    isRepErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isRepErrorVisible = true
+            }
+        },
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt:152]: This is duplicate code. Can we extract this out into a helper function and reuse it?

File: app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+                                        newWeight <= 0 -> "0"
+                                        else -> newWeight.roundToTwoDecimalPlaces().toString()
+                                    }
+                                }
+                            }
+                    }
+                }
+                if (weightValue.toDoubleOrNull() == null) {
+                    isWeightErrorVisible = true
+                }
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt:216]: This entire callback function is identical to the the EditWorkoutOptionalWeightSection Composable inside EditWorkoutSetCalisthenics.kt except for lines 206-209 on this file. Is this check something that was missed in the EditWorkoutSetCalisthenics file? If so, would this be extracted into a helper function and be reused?

**Review by SolidWolf — CHANGES_REQUESTED** [review]

**Comment by Alex-A-Fit** [comment]
I will look to consolidate the dialog logic in the Pr that is after this one

**Pull request merged**